src/euphorie/content/browser/survey.py

```diff
@@ -520,3 +522,51 @@ def tools(self):
                 # we're done if we found at least one measure
                 break
         return tools
+
+
+class ListLinks(BrowserView):
+    attributes_checked = [
+        "description",
+        "introduction",
+        "solution_direction",
+        "legal_reference",
+        "action",
+        "action_plan",
+        "prevention_plan",
+        "requirements",
+    ]
+    url_regex = re.compile(
+        r"https?://[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b"
+        r"(?:[-a-zA-Z0-9()@:%_+.~#?&/=]*)"
+    )
+
+    def extract_links(self, obj):
+        links = []
+        for attrib in self.attributes_checked:
+            value = getattr(aq_base(obj), attrib, "")
+            if value:
+                links.extend(self.url_regex.findall(value))
+        if links:
+            yield {
+                "object": obj,
+                "links": [{"url": link, "text": link} for link in links],
+            }
+        if hasattr(obj, "objectValues"):
+            for child in obj.objectValues():
```

Comment:
Theoretically `objectValues` can contain non-contentish items, for example an object for local workflow policies.  If I want only content I prefer `contentValues`:

```suggestion
        if hasattr(obj, "contentValues"):
            for child in obj.contentValues():
```